fe/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -5074,6 +5125,32 @@ public void replayRenameColumn(TableInfo tableInfo) throws DdlException {
         throw new DdlException("not implmented");
     }
 
+    public void modifyTableDynamicPartition(Database db, OlapTable table, Map<String, String> properties) throws DdlException {
+        Map<String, String> analyzedDynamicPartition = DynamicPartitionUtil.analyzeDynamicPartition(db, table, properties);
+        TableProperty tableProperty = table.getTableProperty();
+        if (tableProperty != null) {
+            tableProperty.modifyTableProperties(analyzedDynamicPartition);
+            dynamicPartitionScheduler.lastUpdateTime = TimeUtils.getCurrentFormatTime();
+            DynamicPartitionInfo info = new DynamicPartitionInfo(db.getId(), table.getId(), table.getTableProperty().getProperties());
+            editLog.logDynamicPartition(info);
+        }
+    }
+
+    public void replayModifyTableDynamicPartition(DynamicPartitionInfo info) {
+        long dbId = info.getDbId();
+        long tableId = info.getTableId();
```

Comment:
If the dbId and tableId are only used here, there is no need to keep it in DynamicPartitionInfo. Because you can log dbId, tableId and DynamicPartitionInfo separated. Or you can implement an struct ModifyTalbeDynamicPartitionJournal which will contain dbId and tableId.

Reply:
I'm confused why can't hold these two ids, and what's the meaning of implement a strcut ModifyTableDynamicPartitionJounal, create a class like JournalEntity?

Reply:
We should ensure that a class contains only necessary information and avoid circular dependency. For example, in this example, unnecessary DB information is coupled in the table information, which will limit the expansion of future functions. Suppose we support the function of moving tables between dB in the future. If there is no record of dbid here, only the information of DB level needs to be modified. If the dbid is recorded here, you need to change the information here, which is completely unnecessary.

Modifytabledynamicpartitionjournal is used to write a log. It is emphasized here that the information recorded in the log is not necessarily the same data structure as the structure existing in the memory.

Reply:
I think you mean that I have no need to hold db in modifyDynamicPartition, I can manipulate the table directly，but I don't konw how to get a table without get it's db first.If I just log table like `editLog.logDynamicPartition(olapTable)`, it violate the meaning of your last sentence, so I think the point is if I just record the table info and modified properties, how can I get the table without get db first, so that I can replay modify properties in the table.

Reply:
What I want to say is that you log 

```
editLog.logDynamicPartition(dbId, tableId, dynamicPartitionInfo)
```

Reply:
@WingsGo Hi, I think @imay misunderstood the meaning of `DynamicPartitionInfo`. `DynamicPartitionInfo` here is just a kind of edit log, which is used to store all information about this "modify" operation, including dbId and tblId.

So please change it back to what you implement before, putting the dbId and tblId inside the 
`DynamicPartitionInfo`, and log as a whole:

`editLog.logDynamicPartition(dynamicPartitionInfo)`

Maybe you can change its class name to "ModifyDynamicPartitionInfo", to make it more like a kind of edit log class.

Reply:
@morningman I have revert and rename "DynamicPartitionInfo" to "ModifyDynamicPartitionInfo"